### PR TITLE
3041 websocket streaming tests

### PR DESCRIPTION
**Pull request by meejah**
This still needs some master-only code from Autobahn.

**Do not merge** until there's a suitable Autobahn release.

This change is [Reviewable](https://reviewable.io/reviews/tahoe-lafs/tahoe-lafs/615)

### CODE REVIEW

**Comment by tpltnt** [comment]
Can you rebase against master to make the CircleCI results meaningful?

**Comment by meejah** [comment]
This is ready to go (assuming CI passes) as of latest autobahn (19.5.2) released recently.

**Comment by tpltnt** [comment]
Looks good to me.

**Comment by tpltnt** [comment]
When can we merge this?

File: src/allmydata/test/test_websocket_logs.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+from autobahn.twisted.testing import create_memory_agent, MemoryReactorClockResolver, create_pumper
+
+from allmydata.web.logs import TokenAuthenticatedWebSocketServerProtocol
+
+
+class TestStreamingLogs(unittest.TestCase):
+    """
+    Test websocket streaming of logs
+
+    Note: depends on un-merged Autobahn branch
```
Comment (exarkun) [inline src/allmydata/test/test_websocket_logs.py]: It's not un-merged now, right?

File: src/allmydata/test/test_websocket_logs.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        self.pumper = create_pumper()
+        self.agent = create_memory_agent(self.reactor, self.pumper, TokenAuthenticatedWebSocketServerProtocol)
+        return self.pumper.start()
+
+    def tearDown(self):
+        return self.pumper.stop()
+
+    @inlineCallbacks
+    def test_one_log(self):
+
```
Comment (exarkun) [inline src/allmydata/test/test_websocket_logs.py]: a docstring would be great :)

**Comment by meejah** [comment]
okay when CI goes green this can be merged.

**Comment by tpltnt** [comment]
The appveyor error
```
[ERROR]
Traceback (most recent call last):
Failure: foolscap.ipb.DeadReferenceError: Connection was lost (to tubid=ncie) (during method=RIUploadHelper.tahoe.allmydata.com:upload)
allmydata.test.test_system.SystemTest.test_upload_and_download_convergent
```
looks negligible to me. Can we merge?

**Comment by meejah** [comment]
Yeah, I'm 99% sure that's one of those "intermittent failures" that keeps happening "sometimes" :/ -- I kicked appveyor to re-run it just to be sure.

**Comment by tpltnt** [comment]
All checks have passed. Merge now?

**Pull request merged**